### ISSUE TITLE
Don't pass `FuturesUnordered` to `join_all`

### ISSUE THREAD

**Issue by Darksonn** [rust-lang/rust-clippy#9842]
### What it does

The `FuturesUnordered` collection is a way to run code concurrently, however it is also possible to use `FuturesUnordered` as a linked-list stack and ignoring its concurrency features. The linked list can be created via `collect` on an iterator or with its `push` method. Then, its `IntoIterator` impl can be used to get the items back out in FILO order like a stack, destroying the `FuturesUnordered` in the progress.

This is what happens when you pass a `FuturesUnordered` to the `join_all` method. The futures are added to the collection, then immediately removed again when `join_all` collects the iterator its given into a `Vec`.

### Lint Name

futures-unordered-with-join-all

### Category

complexity

### Advantage

Any code that passes a `FuturesUnordered` to `join_all` is semantically incorrect and should use a `Vec` instead.

### Drawbacks

None.

### Example

```rust
let tasks = tasks.
    .into_iter()
    .map(|task| my_async_fn(task))
    .collect::<FuturesUnordered<_>>();

let outputs = join_all(tasks).await;
```

Could be written as:

```rust
let tasks = tasks.
    .into_iter()
    .map(|task| my_async_fn(task))
    .collect::<Vec<_>>();

let outputs = join_all(tasks).await;
```
